astropy/nddata/tests/test_utils.py

```diff
@@ -119,6 +119,15 @@ def test_slices_edges():
     assert slc_sm[0].start == slc_sm[0].stop == 0
     assert slc_sm[1].start == slc_sm[1].stop == 0
 
+    with pytest.raises(NoOverlapError):
+        overlap_slices((7, 7), (0, 0), (-2, -2))
+    with pytest.raises(NoOverlapError):
+        overlap_slices((7, 7), (3, 3), (-2, -2))
+    with pytest.raises(NoOverlapError):
+        overlap_slices((7, 7), (3, 3), (-2, 2))
+    with pytest.raises(NoOverlapError):
+        overlap_slices((7, 7), (3, 3), (2, -2))
```

Comment:
```suggestion

@pytest.mark.parametrize(
    "inputs",
    [
        ((7, 7), (0, 0), (-2, -2)),
        ((7, 7), (3, 3), (-2, -2)),
        ((7, 7), (3, 3), (-2, 2)),
        ((7, 7), (3, 3), (2, -2)),
    ],
 )
def test_slices_no_overlap(inputs):
    with pytest.raises(NoOverlapError):
        overlap_slices(*inputs)
```

Reply:
I discovered *after* I wrote this suggestion that there's already a test with the exact same name and structure I suggested to instead of applying my suggestion as is, pleas move the new test cases there.

Reply:
Done!